spec/emerald/rubify_spec.rb

```diff
@@ -0,0 +1,131 @@
+      compiled_code = Emerald::Rubify.new([Emerald::String.new('"Hey there"')]).rubify
+      expect(compiled_code).to eq('"Hey there"')
+    end
+
+    it "generates code from two strings" do
+      compiled_code = Emerald::Rubify.new([Emerald::String.new('"goat"'),Emerald::String.new('"duck"')]).rubify
+      expect(compiled_code).to eq('"goat" "duck"')
+    end
+  end
+
+  describe "generates numeric operations" do
+    it "generates addition" do
+      compiled_code = Emerald::Rubify.new([Emerald::Atom.new('+'), Emerald::Number.new(1.0), Emerald::Number.new(2.0)]).rubify
+      expect(compiled_code).to eq('1.0 + 2.0')
+    end
+
+    it "generates addition with several arguments" do
+      compiled_code = Emerald::Rubify.new([Emerald::Atom.new('+'), Emerald::Number.new(1.0), Emerald::Number.new(2.0), Emerald::Number.new(9),  Emerald::Number.new(0.5)]).rubify
+      expect(compiled_code).to eq('1.0 + 2.0 + 9 + 0.5')
+    end
+
+    it "generates subtration operations" do
+      compiled_code = Emerald::Rubify.new([Emerald::Atom.new('-'), Emerald::Number.new(1.0), Emerald::Number.new(2.0)]).rubify
+      expect(compiled_code).to eq('1.0 - 2.0')
+    end
+
+    it "generates division operations" do
+      compiled_code = Emerald::Rubify.new([Emerald::Atom.new('/'), Emerald::Number.new(1.0), Emerald::Number.new(2.0)]).rubify
+      expect(compiled_code).to eq('1.0 / 2.0')
+    end
+
+    it "generates multiplication" do
+      compiled_code = Emerald::Rubify.new([Emerald::Atom.new('*'), Emerald::Number.new(1.0), Emerald::Number.new(2.0)]).rubify
+      expect(compiled_code).to eq('1.0 * 2.0')
+    end
+
+    it "generates less than operations" do
+      compiled_code = Emerald::Rubify.new([Emerald::Atom.new('<'), Emerald::Number.new(1.0), Emerald::Number.new(2.0)]).rubify
+      expect(compiled_code).to eq('1.0 < 2.0')
+    end
+
+    it "generates greater than operations" do
+      compiled_code = Emerald::Rubify.new([Emerald::Atom.new('>'), Emerald::Number.new(1.0), Emerald::Number.new(2.0)]).rubify
+      expect(compiled_code).to eq('1.0 > 2.0')
+    end
+
+    it "generates less than or equal to operations" do
+      compiled_code = Emerald::Rubify.new([Emerald::Atom.new('<='), Emerald::Number.new(1.0), Emerald::Number.new(2.0)]).rubify
+      expect(compiled_code).to eq('1.0 <= 2.0')
+    end
+
+    it "generates greater than or equal to operations" do
+      compiled_code = Emerald::Rubify.new([Emerald::Atom.new('>='), Emerald::Number.new(1.0), Emerald::Number.new(2.0)]).rubify
+      expect(compiled_code).to eq('1.0 >= 2.0')
+    end
+
+    #it "generates addition with negative arguments" do
+      #compiled_code = Emerald::Rubify.new([Emerald::Atom.new('+'), Emerald::Number.new(1.0), Emerald::Number.new(2.0), Emerald::Number.new(-9)]).rubify
+      #expect(compiled_code).to eq('1.0 + 2.0 - 9')
+    #end
+  end
+
+  describe "generates code for logical operations" do
+    it "generates code for querying nil on a number" do
```

Comment:
In Lisp it's only a function call an atom is in a List, so this would instead compile to this:

```ruby
:"nil?"; 1.0
```